### PR TITLE
moved utilization calculation to the SDK

### PR DESCRIPTION
**Pull request by smartdevcom**

### CODE REVIEW

File: packages/ui/hooks/useFusePoolData.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ export const useFusePoolData = (poolId: string) => {
 
       const res = await fuse.fetchFusePoolData(poolId, address);
       const assetsWithPrice: MarketData[] = [];
+      console.log(res.assets);
```
Comment (peetzweg) [inline packages/ui/hooks/useFusePoolData.ts:35]:
```suggestion

```

File: packages/ui/components/pages/Fuse/FusePoolPage/index.tsx [hunk 2]
```diff
@@ -112,14 +112,7 @@ const FusePoolPage = memo(() => {
...
                   label="Utilization"
-                  value={
-                    data
-                      ? data.totalSuppliedFiat.toString() === '0'
-                        ? '0%'
-                        : ((data?.totalBorrowedFiat / data?.totalSuppliedFiat) * 100).toFixed(2) +
-                          '%'
-                      : undefined
-                  }
+                  value={data ? data.utilization.toFixed(2) + '%' : undefined}
```
Comment (peetzweg) [inline packages/ui/components/pages/Fuse/FusePoolPage/index.tsx:115]: this pr does not seem to contain the code which is setting the `data.utilization` value. Was it already available all along?

File: packages/sdk/src/modules/FusePools.ts [hunk 3]
```diff
@@ -96,13 +99,25 @@ export function withFusePools<TBase extends FuseBaseConstructor>(Base: TBase) {
...
 
         asset.liquidityNative =
           Number(utils.formatUnits(asset.liquidity)) * Number(utils.formatUnits(asset.underlyingPrice));
 
         totalLiquidityNative += asset.liquidityNative;
+
+        if (!asset.isSupplyPaused) suppliedForUtilization += asset.totalSupplyNative;
+        if (!asset.isBorrowPaused) borrowedForUtilization += asset.totalBorrowNative;
+        if (suppliedForUtilization !== 0) {
+          utilization = (borrowedForUtilization / suppliedForUtilization) * 100;
```
Comment (peetzweg) [inline packages/sdk/src/modules/FusePools.ts:119]: This calculation can be done outside of the for loop.

Comment (peetzweg) [reply packages/sdk/src/modules/FusePools.ts]: But maybe this PR is more about moving hte calculation outside the FE and into SDK and not about this ticket here yet, right?

https://github.com/Midas-Protocol/monorepo/issues/184

Comment (smartdevcom) [reply packages/sdk/src/modules/FusePools.ts]: Yes but also this PR fixed `utilization` calculation based on only borrow-able assets

Comment (smartdevcom) [reply packages/sdk/src/modules/FusePools.ts]: makes sense

Comment (smartdevcom) [reply packages/sdk/src/modules/FusePools.ts]: > Also not sure if this calculation is correct. thinking
> 
> Shouldn't it be something like `currently borrowed / total amount which is abled to borrowed` ?
> 
> Where as `total amount which is abled to borrowed` is `sum of all the supply of borrowable assets`, so not including the supply of assets which are not borrowable.
> 
> `currently borrowed` => `sum of all borrowed assets`

Check this https://github.com/Midas-Protocol/monorepo/pull/201/commits/a965ecbd72e6a7a9c5f866f261e5004f600463c5

Comment (peetzweg) [reply packages/sdk/src/modules/FusePools.ts]: Perfect, I think this should work as intended in #184

**Review by peetzweg — APPROVED** [review]

**Pull request merged**